middlewares/token.js

```diff
@@ -0,0 +1,34 @@
+const jwt = require("jsonwebtoken");
+const { User } = require("../models/user");
+
+function auth(req, res, next) {
+  const authHeader = req.headers.authorization || "";
+  const [bearer, token] = authHeader.split(" ", 2);
+
+  if (bearer !== "Bearer") {
+    return res.status(401).send({ message: "Not authorized" });
+  }
+  jwt.verify(token, process.env.JWT_SECRET, async (err, decode) => {
+    if (err) {
```

Comment:
```suggestion
    if (err && err.name ...) {
```
уникайте непотрібної вкладеності